File: go-app/example/create.go
package example

import (
	"bytes"
	"context"
	"encoding/json"
	"fmt"
	"github.com/elastic/go-elasticsearch/v8"
	"go-app/types"
	"io"
	"log"
)

func CreateNewUser(es *elasticsearch.Client, user types.User, friends []string) {
	jsonData, err := json.Marshal(user)
	if err != nil {
		log.Fatalf("Error marshalling user data: %s", err)
	}

	response, err := es.Index(
		"users",
		bytes.NewReader(jsonData),
		es.Index.WithContext(context.Background()),
		es.Index.WithRefresh("true"),
	)
	if err != nil {
		log.Fatalf("Error creating new user: %s", err)
	}
	defer func(body io.ReadCloser) {
		_ = body.Close()
	}(response.Body)

	fmt.Println("New user created:", response.Status())

	friendship := types.Friendship{
		User:    user.Name,
		Friends: friends,
	}
	jsonFriendship, err := json.Marshal(friendship)
	if err != nil {
		log.Fatalf("Error marshalling friendship data: %s", err)
	}

	friendshipResponse, err := es.Index(
		"friendships",
		bytes.NewReader(jsonFriendship),
		es.Index.WithContext(context.Background()),
		es.Index.WithRefresh("true"),
	)
	if err != nil {
		log.Fatalf("Error creating friendship: %s", err)
	}
	defer func(body io.ReadCloser) {
		_ = body.Close()
	}(friendshipResponse.Body)

	fmt.Println("Friendship created:", friendshipResponse.Status())
}

File: nodejs-app/create.js
const client = require('./config');

const usersIndex = 'users';
const friendshipsIndex = 'friendships';

async function createNewUser(data) {
    try {
        const response = await client.index({
            index: usersIndex,
            document: data,
            refresh: true,
        });

        console.log(`New user created:`, response);
    } catch (error) {
        console.error('Error creating new user:', error.meta?.body || error);
    }
}

async function createNewFriendship(data) {
    try {
        const response = await client.index({
            index: friendshipsIndex,
            document: data,
            refresh: true,
        });

        console.log(`New friendship created:`, response);
    } catch (error) {
        console.error('Error creating new friendship:', error.meta?.body || error);
    }
}

(async () => {
    await createNewUser({
        name: 'Frank',
        age: 29,
        hobbies: ['gaming', 'traveling'],
    });
    await createNewFriendship({
        user: 'Frank',
        friends: ['Eve', 'Dave'],
    });
})();
